api/controllers/pr/index.js

```diff
@@ -23,32 +23,57 @@ exports.index = (req, res) => {
   Promise.all([
     findGithubPrs(github, username),
     findGitlabPrs(gitlab, username),
-    github.users.getByUsername({ username }).then(logCallsRemaining),
+    searchGithubUser(github, username),
+    searchGitlabUser(gitlab, username),
   ])
-    .then(([prs, mrs, user]) => {
-      if (user.data.type !== 'User') {
+    .then(([prs, mrs, user, gitlab_user]) => {
+      let isGitHubUser = true;
+      if (user == null) isGitHubUser = false;
+      let isGitLabUser = true;
+      if (JSON.stringify(gitlab_user) == JSON.stringify([]))
+        isGitLabUser = false;
+
+      if (
+        (!isGitHubUser || (isGitHubUser && user.data.type !== 'User')) &&
+        !isGitLabUser
+      ) {
         return Promise.reject('notUser');
       }
 
       // Combine github PRs with the gitlab MRs in sorted order.
       // Most recent PRs/MRs will come first.
-      prs = prs.concat(mrs).sort((pr1, pr2) => {
-        const date1 = moment(pr1.created_at, 'MMMM Do YYYY');
-        const date2 = moment(pr2.created_at, 'MMMM Do YYYY');
-        if (date1.isSame(date2)) return 0;
-        else if (date1.isAfter(date2)) return -1;
-        return 1;
-      });
+      if (prs) {
+        prs = prs.concat(mrs).sort((pr1, pr2) => {
+          const date1 = moment(pr1.created_at, 'MMMM Do YYYY');
+          const date2 = moment(pr2.created_at, 'MMMM Do YYYY');
+          if (date1.isSame(date2)) return 0;
+          else if (date1.isAfter(date2)) return -1;
+          return 1;
+        });
+      }
 
       // TODO: If user is empty, try looking them up on gitlab.
+      if (!isGitHubUser && isGitLabUser) {
+        user = gitlab_user;
+      }
+
+      if (isGitHubUser) {
+        const data = {
+          prs,
+          username,
+          userImage: user.data.avatar_url,
+        };
 
-      const data = {
-        prs,
-        username,
-        userImage: user.data.avatar_url,
-      };
+        res.json(data);
+      } else if (isGitLabUser) {
+        const data = {
+          mrs,
```

Comment:
I see why you surrounded the `prs.concat()` thing in an `if` now. Is it necessary? Can we just make `prs` be an empty array if github user doesn't exist and then just concat as usual? Can always just pass `prs` then and simplify this a little so only the `userImage` is conditional, we can use a ternary or something then?

Reply:
Nice suggestion. Will try to implement this.